pyproject.toml

```diff
@@ -17,20 +17,21 @@ dependencies = [
   "optax>=0.1.4",
   "scipy>=1.10.0",
   "jaxopt>=0.8.5",
+  "jaxtyping>=0.3.3",
   "equinox>=0.12.2",
   "optimistix>=0.0.10",
 ]
-version = "0.4.0"
+version = "0.5.0"
 
 [project.optional-dependencies]
-dev = ["mypy>=0.991", "pytest>=7.2.0", "ruff==0.9.0", "pre-commit>=4.0.0", "matplotlib>=3.10.0"]
+dev = ["mypy>=0.991", "pytest>=7.2.0", "ruff==0.11.12", "pre-commit>=4.0.0", "matplotlib>=3.10.0"]
```

Comment:
Ruff is already on 0.14, but we can fix that later. Also have to make sure this aligns with the ruff version in the CI.